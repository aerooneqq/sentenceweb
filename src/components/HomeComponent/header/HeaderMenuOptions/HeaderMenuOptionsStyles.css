@media screen and (max-width: 560px){ 
    #menuOptionsContainer{
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        background: red;
        
        height: 0px;
        width: 100%;

        opacity: 0;
    }

    .showMenuAnimation{ 
        animation: showMenu 0.5s ease-out forwards;
    }

    @keyframes showMenu { 
        from { 
            opacity: 0;
            height: 0px;
        }

        to { 
            opacity: 1;
            height: 160px;
        }
    }

    .hideMenuAnimation { 
        animation: hideMenu 0.5s ease-in forwards;
    }

    @keyframes hideMenu { 
        from { 
            opacity: 1;
            height: 160px;
        }

        to { 
            opacity: 0;
            height: 0px;
        }
    }
}

@media screen and (min-width: 560px){ 
    #menuOptionsContainer{
        display: flex;
        flex-direction: row;
        align-items: center;
    }
}