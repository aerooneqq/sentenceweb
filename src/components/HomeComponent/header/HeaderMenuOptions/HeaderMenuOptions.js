import React, {Component, lazy} from "react";

//Styles
import "./HeaderMenuOptionsStyles.css"

//Components
const HeaderMenuItem = lazy(() => import("../HeaderMenuItem/HeaderMenuItem"));

export default class HeaderMenuOptions extends Component{ 
    menuOptionsModels = [{ ID: 0, Name: "About", Href: "" },
    { ID: 1, Name: "Examples", Href: "" }, { ID: 2, Name: "Try", Href: "" },
    { ID: 3, Name: "Contacts", Href: "" }]

    constructor(props){ 
        super(props);
    }

    getMenuOptionsList() {
        let menuOptions = []

        for (var i = 0; i < this.menuOptionsModels.length; i++) {
            menuOptions.push(<HeaderMenuItem menuItem={this.menuOptionsModels[i]} />)
        }

        return menuOptions
    }

    render() { 
        let jsxMenuOptions = this.getMenuOptionsList();

        return ( 
            <div id="menuOptionsContainer">
                {jsxMenuOptions}
            </div>
        );
    }   
}