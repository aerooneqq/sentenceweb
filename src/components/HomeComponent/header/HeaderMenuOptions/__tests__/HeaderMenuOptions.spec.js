import React from 'react'
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import HeaderMenuOptions from '../HeaderMenuOptions';

describe("Header menu options", () => { 
    it("should render correctly" , () => { 
        const headerMenuOptions = shallow(<HeaderMenuOptions />);

        expect(shallowToJson(headerMenuOptions)).toMatchSnapshot();
    });
});

