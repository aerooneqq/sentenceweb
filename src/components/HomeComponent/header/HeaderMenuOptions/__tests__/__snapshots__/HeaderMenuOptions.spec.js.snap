// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header menu options should render correctly 1`] = `
<div
  id="menuOptionsContainer"
>
  <lazy
    key="0"
    menuItem={
      Object {
        "Href": "",
        "ID": 0,
        "name": "About",
      }
    }
  />
  <lazy
    key="1"
    menuItem={
      Object {
        "Href": "",
        "ID": 1,
        "name": "Examples",
      }
    }
  />
  <lazy
    key="2"
    menuItem={
      Object {
        "Href": "",
        "ID": 2,
        "name": "Try",
      }
    }
  />
  <lazy
    key="3"
    menuItem={
      Object {
        "Href": "",
        "ID": 3,
        "name": "Contacts",
      }
    }
  />
</div>
`;
